x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java

```diff
@@ -2900,9 +2987,25 @@ private static void restOrTransportOrigin(
     }
 
     private static void authentication(Authentication authentication, MapBuilder<String, String> checkedFields) {
+        assertThat("use remoteAccessAuthentication instead", authentication.isRemoteAccess(), is(false));
+        putCheckedFieldsForAuthentication(authentication, checkedFields);
+    }
+
+    private static void remoteAccessAuthentication(
+        Authentication authentication,
+        MapBuilder<String, String> checkedFields,
+        MapBuilder<String, String> checkedLiteralFields,
+        String expectedRemoteAccessLiteralField
+    ) {
+        assertThat("authentication must be remote access", authentication.isRemoteAccess(), is(true));
+        putCheckedFieldsForAuthentication(authentication, checkedFields);
+        checkedLiteralFields.put(LoggingAuditTrail.REMOTE_ACCESS_FIELD_NAME, expectedRemoteAccessLiteralField);
```

Comment:
Good call, I've moved building inside this method. The original thinking of passing the string was that building the expected string only works for a subset of authentication instances (e.g., run-as is not supported), and we limit which authentication instance we use in the test so I wanted to keep choosing a valid authentication instance and building the expected string in the same method. But I think it's fair enough to move string building into this method since if we pass an unexpected authentication instance, we will get a clear error message.